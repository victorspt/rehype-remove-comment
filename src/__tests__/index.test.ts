import { describe, expect, test } from '@jest/globals';
import rehypeRaw from 'rehype-raw';
import rehypeRemoveComment from '../index.ts';
// @ts-expect-error The next dependency has unrecognized export.
import { serialize } from 'next-mdx-remote/serialize';

const mdContent =
  '---\n' +
  'title: Test Title\n' +
  'layout: testLayout.hbs\n' +
  '---\n' +
  '\n' +
  '<!-- HTML comment. -->\n' +
  '\n' +
  '<!--\n' +
  '\n' +
  'Multiline HTML comment.\n' +
  '\n' +
  'Testing.\n' +
  '\n' +
  'Testing more.\n' +
  '\n' +
  '-->\n' +
  '\n' +
  '---\n' +
  '\n' +
  '# Level-1 heading\n' +
  '\n' +
  '## Level-2 heading\n' +
  '\n' +
  'Markdown paragraph. Testing `inline code`.\n' +
  '\n' +
  '<section class="note">\n' +
  '### Level-3 heading inside HTML\n' +
  '  <p>Paragraph inside &lt;section&gt;</p>\n' +
  '</section>\n' +
  '\n' +
  '<!--\n' +
  'HTML multiline comment.\n' +
  '\n' +
  '```javascript\n' +
  "import { numbersUrl } from './constants.js';\n" +
  "import { mapNumber, isValid } from './functions.js';\n" +
  '\n' +
  '// Javascript line comment inside HTML multiline comment. "Test."\n' +
  '\n' +
  '/**\n' +
  ' * Javascript JSdoc inside HTML comment.\n' +
  ' */\n' +
  'async function getResult() {\n' +
  '  const response = await fetch(numbersUrl);\n' +
  '  const numbers = await response.json();\n' +
  '  const result = numbers\n' +
  '    .map(mapNumber)\n' +
  '    .reduce((accu /* Javascript inline comment. */, curr) => {\n' +
  '      return (isValid`${curr}`) ? accu + curr : accu;\n' +
  '    });\n' +
  '  return result;\n' +
  '}\n' +
  '```\n' +
  '\n' +
  '```python\n' +
  'from db import db\n' +
  '\n' +
  '# Python line comment inside HTML comment.\n' +
  '\n' +
  'def sendNumbers():\n' +
  '    """\n' +
  '    Python multiline comment inside HTML comment.\n' +
  '    Testing.\n' +
  '    """\n' +
  '    return db.getNumbers()\n' +
  '```\n' +
  '\n' +
  '  <!-- HTML line comment inside an HTML multiline comment. -->\n' +
  '\n' +
  '-->\n' +
  '\n' +
  '## Level-2 heading\n' +
  '\n' +
  'Markdown Paragraph.\n' +
  '\n' +
  '- Markdown list item 1.\n' +
  '- Markdown list item 2.\n' +
  '- Markdown list item 3.\n' +
  '- Markdown list item 4.\n' +
  '\n' +
  '<table class="table-test">\n' +
  '  <tr>\n' +
  '    <td>HTML table cell.</td>\n' +
  '  </tr>\n' +
  '</table>\n' +
  '\n' +
  '<!-- HTML comment. -->\n' +
  '\n' +
  '[Markdown link](https://example.com)\n';

const optionsRemoveComments = {
  mdxOptions: {
    format: 'md',
    rehypePlugins: [rehypeRaw, rehypeRemoveComment],
  },
};

const optionsPreserveComments = {
  mdxOptions: {
    format: 'md',
    rehypePlugins: [rehypeRaw],
  },
};

describe('The rehype-remove-comment plugin', () => {
  test('removes HTML comments from the md file', async () => {
    const { compiledSource } = await serialize(
      mdContent,
      optionsRemoveComments,
    );
    expect(compiledSource).toMatchSnapshot();
  });

  test('when not applied preserve the HTML comments in the md file', async () => {
    const { compiledSource } = await serialize(
      mdContent,
      optionsPreserveComments,
    );
    expect(compiledSource).toMatchSnapshot();
  });
});
