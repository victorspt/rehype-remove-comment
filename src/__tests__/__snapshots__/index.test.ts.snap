// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The rehype-remove-comment plugin removes HTML comments from the md file 1`] = `
"/*@jsxRuntime automatic @jsxImportSource react*/
const {Fragment: _Fragment, jsx: _jsx, jsxs: _jsxs} = arguments[0];
const {useMDXComponents: _provideComponents} = arguments[0];
function _createMdxContent(props) {
  const _components = Object.assign({
    hr: "hr",
    h2: "h2",
    h1: "h1",
    p: "p",
    code: "code",
    section: "section",
    ul: "ul",
    li: "li",
    table: "table",
    tbody: "tbody",
    tr: "tr",
    td: "td",
    a: "a"
  }, _provideComponents(), props.components);
  return _jsxs(_Fragment, {
    children: [_jsx(_components.hr, {}), "\\n", _jsx(_components.h2, {
      children: "title: Test Title\\nlayout: testLayout.hbs"
    }), "\\n", "\\n", "\\n", _jsx(_components.hr, {}), "\\n", _jsx(_components.h1, {
      children: "Level-1 heading"
    }), "\\n", _jsx(_components.h2, {
      children: "Level-2 heading"
    }), "\\n", _jsxs(_components.p, {
      children: ["Markdown paragraph. Testing ", _jsx(_components.code, {
        children: "inline code"
      }), "."]
    }), "\\n", _jsxs(_components.section, {
      className: "note",
      children: ["\\n### Level-3 heading inside HTML\\n  ", _jsx(_components.p, {
        children: "Paragraph inside <section>"
      }), "\\n"]
    }), "\\n", "\\n", _jsx(_components.p, {
      children: "-->"
    }), "\\n", _jsx(_components.h2, {
      children: "Level-2 heading"
    }), "\\n", _jsx(_components.p, {
      children: "Markdown Paragraph."
    }), "\\n", _jsxs(_components.ul, {
      children: ["\\n", _jsx(_components.li, {
        children: "Markdown list item 1."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 2."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 3."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 4."
      }), "\\n"]
    }), "\\n", _jsxs(_components.table, {
      className: "table-test",
      children: ["\\n  ", _jsx(_components.tbody, {
        children: _jsxs(_components.tr, {
          children: ["\\n    ", _jsx(_components.td, {
            children: "HTML table cell."
          }), "\\n  "]
        })
      })]
    }), "\\n", "\\n", _jsx(_components.p, {
      children: _jsx(_components.a, {
        href: "https://example.com",
        children: "Markdown link"
      })
    })]
  });
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? _jsx(MDXLayout, Object.assign({}, props, {
    children: _jsx(_createMdxContent, props)
  })) : _createMdxContent(props);
}
return {
  default: MDXContent
};
"
`;

exports[`The rehype-remove-comment plugin when not applied preserve the HTML comments in the md file 1`] = `
"/*@jsxRuntime automatic @jsxImportSource react*/
/*HTML comment.*/
/*

Multiline HTML comment.

Testing.

Testing more.

*/
/*
HTML multiline comment.

\`\`\`javascript
import { numbersUrl } from './constants.js';
import { mapNumber, isValid } from './functions.js';

// Javascript line comment inside HTML multiline comment. "Test."

/**
* Javascript JSdoc inside HTML comment.
*/
async function getResult() {
const response = await fetch(numbersUrl);
const numbers = await response.json();
const result = numbers
.map(mapNumber)
.reduce((accu /* Javascript inline comment. */, curr) => {
return (isValid\`\${curr}\`) ? accu + curr : accu;
});
return result;
}
\`\`\`

\`\`\`python
from db import db

# Python line comment inside HTML comment.

def sendNumbers():
"""
Python multiline comment inside HTML comment.
Testing.
"""
return db.getNumbers()
\`\`\`

<!-- HTML line comment inside an HTML multiline comment.*/
/*HTML comment.*/
const {Fragment: _Fragment, jsx: _jsx, jsxs: _jsxs} = arguments[0];
const {useMDXComponents: _provideComponents} = arguments[0];
function _createMdxContent(props) {
  const _components = Object.assign({
    hr: "hr",
    h2: "h2",
    h1: "h1",
    p: "p",
    code: "code",
    section: "section",
    ul: "ul",
    li: "li",
    table: "table",
    tbody: "tbody",
    tr: "tr",
    td: "td",
    a: "a"
  }, _provideComponents(), props.components);
  return _jsxs(_Fragment, {
    children: [_jsx(_components.hr, {}), "\\n", _jsx(_components.h2, {
      children: "title: Test Title\\nlayout: testLayout.hbs"
    }), "\\n", "\\n", "\\n", _jsx(_components.hr, {}), "\\n", _jsx(_components.h1, {
      children: "Level-1 heading"
    }), "\\n", _jsx(_components.h2, {
      children: "Level-2 heading"
    }), "\\n", _jsxs(_components.p, {
      children: ["Markdown paragraph. Testing ", _jsx(_components.code, {
        children: "inline code"
      }), "."]
    }), "\\n", _jsxs(_components.section, {
      className: "note",
      children: ["\\n### Level-3 heading inside HTML\\n  ", _jsx(_components.p, {
        children: "Paragraph inside <section>"
      }), "\\n"]
    }), "\\n", "\\n", _jsx(_components.p, {
      children: "-->"
    }), "\\n", _jsx(_components.h2, {
      children: "Level-2 heading"
    }), "\\n", _jsx(_components.p, {
      children: "Markdown Paragraph."
    }), "\\n", _jsxs(_components.ul, {
      children: ["\\n", _jsx(_components.li, {
        children: "Markdown list item 1."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 2."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 3."
      }), "\\n", _jsx(_components.li, {
        children: "Markdown list item 4."
      }), "\\n"]
    }), "\\n", _jsxs(_components.table, {
      className: "table-test",
      children: ["\\n  ", _jsx(_components.tbody, {
        children: _jsxs(_components.tr, {
          children: ["\\n    ", _jsx(_components.td, {
            children: "HTML table cell."
          }), "\\n  "]
        })
      })]
    }), "\\n", "\\n", _jsx(_components.p, {
      children: _jsx(_components.a, {
        href: "https://example.com",
        children: "Markdown link"
      })
    })]
  });
}
function MDXContent(props = {}) {
  const {wrapper: MDXLayout} = Object.assign({}, _provideComponents(), props.components);
  return MDXLayout ? _jsx(MDXLayout, Object.assign({}, props, {
    children: _jsx(_createMdxContent, props)
  })) : _createMdxContent(props);
}
return {
  default: MDXContent
};
"
`;
